lldb/source/Plugins/Process/elf-core/ProcessElfCore.cpp

```diff
@@ -272,6 +273,13 @@ Status ProcessElfCore::DoLoadCore() {
       }
     }
   }
+  llvm::Expected<std::shared_ptr<ProcessElfGpuCore>> gpu_process =
```

Comment:
99% of the time we won't have a GPU core in the core file. What errors would we report here? There shouldn't be an error if there is no GPU core, but that isn't clear from the return value `llvm::Expected<std::shared_ptr<ProcessElfGpuCore>>` as this says "I expect to get something valid, and if I don't you should report the error.  But this error has nothing to do with the native core file. We should check for embedded core files with something like:
```
ProcessElfCore::LoadEmbeddedCores();
```
And the error should have no say if the loading of the native core file is ok. The GPU core files can emit their own errors if there is some mismatch or problem, but this class shouldn't care other than just trying to load them.

Reply:
Makes sense. Made the change.